### ISSUE TITLE
Please, make me understanding callback system.

### ISSUE THREAD

**Issue by StigmaKim** [Automattic/mongoose#2985]
Sorry, I`m in trouble with callback function operation.

 my code is below

---------------------------------

var query = member.where({ member : Member.member });
var output= '';
var Grouparr=[];
	
query.find({}, function(err, docs){
	if(!err){
		for(var a in docs){
			var qu = group.where({ _id : docs[a].groupid }); // 
			qu.findOne(function(err, docu){
				if(!err){
					Grouparr.push(docu);
				}
			});
		};
	}
});
	
console.log("array : " + Grouparr);
res.send(Grouparr);

--------------------------
I checked that Grouparr.push(docu) work, and docu has correct data.
But when console.log("array : " + Grouparr); is called, Grouparr is null.
As I think, this is because console.log is called earlier than complete of findOne`s callback function.

In other words, I want to execute some lines after callback function complete, 
How can I do it T.T

If my prediction is right, plz fix my code.
If anyone fixes my code, maybe I can understand how do I use callback system from that.

Please Help me...

**Comment by StigmaKim** [comment]
I found async module.
sorry for a little thing makes big problem.
